src/clr-angular/progress/spinner/spinner.ts

```diff
@@ -3,41 +3,80 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, Input } from '@angular/core';
+import { Component, Input, HostBinding } from '@angular/core';
 import { isBooleanAttributeSet } from '../../utils/component/is-boolean-attribute-set';
 
-const SPINNER_BASE_CLASS = 'spinner';
-
-const SPINNER_INVERSE = 'spinner-inverse';
-const SPINNER_INLINE = 'spinner-inline';
-
-const SPINNER_SMALL_SIZE = 'spinner-sm';
-const SPINNER_MEDIUM_SIZE = 'spinner-md';
-
 @Component({
   selector: 'clr-spinner',
   template: `
     <ng-content></ng-content>
   `,
   host: {
-    '[attr.class]': 'assignClass',
     '[attr.aria-live]': 'setAriaLive',
     '[attr.aria-busy]': 'true',
   },
 })
 export class ClrSpinner {
+  @HostBinding('class.spinner')
+  get spinnerClass() {
+    return true;
+  }
+
   // Style
-  @Input() inline: boolean;
-  @Input() inverse: boolean;
+  private _inline: boolean;
+  @HostBinding('class.spinner-inline')
+  get inlineClass() {
+    return this._inline;
+  }
+
+  @Input('clrInline')
+  set clrInline(value: boolean | string) {
+    this._inline = isBooleanAttributeSet(value);
+  }
+
+  private _inverse: boolean;
+  @HostBinding('class.spinner-inverse')
+  get inverseClass() {
+    return this._inverse;
+  }
+
+  @Input('clrInverse')
+  set clrInverse(value: boolean | string) {
+    this._inverse = isBooleanAttributeSet(value);
+  }
 
   // Size
-  @Input() small: boolean;
-  @Input() medium: boolean;
+  private _small: boolean;
+  @HostBinding('class.spinner-sm')
+  get smallClass() {
+    return this._small;
+  }
+
+  @Input('clrSmall')
+  set clrSmall(value: boolean | string) {
+    this._small = isBooleanAttributeSet(value);
+  }
+
+  private _medium: boolean;
+  @HostBinding('class.spinner-md')
+  get mediumClass() {
+    // You could have only one set
+    if (this._small) {
+      return false;
```

Comment:
So the behavior is when `clrSmall` is on the spinner, it overrides the `clrMedium`? if that's the case, you need to make `this._medium` to `false` when `this._small` becomes `true`.

Reply:
And please add tests for this behavior.